docs/understanding-packages.md

```diff
@@ -0,0 +1,228 @@
+`node_modules/frock-static` and `./mocks/mock-api-handler` will resolve to the
+file `./mocks/mock-api-handler.js`.
+
+## `frockfile.json` Resolution
+
+When you run `frock`, it will search for a `frockfile.json` from the current
+directory upward; the process is the same as `npm` searching for a
+`package.json`. Just as with `npm`, when it finds a frockfile it will use the
+frockfile's parent directory as its working directory, performing all module
+resolution from that directory.
+
+## Using `frock` in your projects
+
+`frock` is meant to be included in your packages as a dev-dependency, similar
+to a `grunt` or `gulp` setup. Continuing on from the `my-project` example above,
+your `package.json` would look something like this:
+
+```json
+{
+  "name": "my-project",
+  "version": "0.1.0",
+  "description": "My awesome project.",
+  "scripts": {
+    "frock": "frock"
+  },
+  "devDependencies": {
+    "frock": "^1.0.1",
+    "frock-static": "^1.0.0"
+  },
+  "dependencies": {}
+}
+```
+
+This will allow local development to be done using `frock`, but without
+installing those dependencies when your project is in production. Additionally,
+the `frock` entry in the `scripts` key will allow you to run frock even if your
+`PATH` isn't set as recommended above using an `npm run frock`.
+
+## Using `frock` with optional packages
+
+`frock` intentionally has a small core and the smallest possible dependency set,
+but there are some additional packages that you can use to enhance its
+functionality.
+
+### Databases
+
+If the [level][] package is present, you can access a `leveldb` instance(s) in
+your plugins. Continuing from the examples above, your `package.json` would look
+as follows:
+
+```json
+{
+  "name": "my-project",
+  "version": "0.1.0",
+  "description": "My awesome project.",
+  "scripts": {
+    "frock": "frock"
+  },
+  "devDependencies": {
+    "frock": "^1.0.1",
+    "frock-static": "^1.0.0",
+    "level": "^1.4.0"
+  },
+  "dependencies": {}
+}
+```
+
+Then define a database path in your `package.json`:
+
+```json
+{
+  "db": {
+    "path": "_db"
+  },
+  "servers": [
+    {
+      "port": 8080,
+      "routes": [
+        {
+          "path": "/api/",
+          "methods": "any",
+          "handler": "./mocks/mock-api-handler"
+        },
+        {
+          "path": "*",
+          "methods": "any",
+          "handler": "frock-static",
+          "options": {
+            "dir": "./static"
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+For an example of a plugin that uses a database, see the [db example][].
+
+*Note:* trying to access a database when the `level` or `levelup` packages are
+not locally installed will cause an error to be thrown. Not having a database
+path defined will similarly throw an error.
+
+### Javascript Compilers/Transpilers
+
+In local mocks—that is, mocks that are in the local project directory—you may
+want to write in newer dialects of Javascript (like ES2015) or TypeScript;
```

Comment:
["JavaScript"](http://programmers.stackexchange.com/questions/135905/legal-ramifications-of-use-of-the-javascript-trademark)